src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3050,6 +3066,12 @@ void addProperties(ImmutableArray<ParameterSymbol> recordParameters)
                         builder.InstanceInitializersForRecordDeclarationWithParameters.Insert(addedCount, new FieldOrPropertyInitializer.Builder(property.BackingField, paramList.Parameters[param.Ordinal]));
                         addedCount++;
                     }
+                    else if (existingMember.Kind switch { SymbolKind.Property => true, SymbolKind.Field => true, _ => false })
+                    {
+                        // There already exists a member corresponding to the candidate synthesized property.
+                        // The deconstructor is specified to simply assign from this property to the corresponding out parameter.
+                        existingOrAddedMembers.Add(existingMember);
+                    }
```

Comment:
>} [](start = 20, length = 1)

else? What about an event for example?

Reply:
Added an else clause which adds an error.